eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/EventCriteria.java

```diff
@@ -57,158 +89,94 @@ static EventCriteria anyEvent() {
+     *      <li>COURSE -> X</li>
      * </ul>
      * The following events will match:
      * <ul>
      *     <li> Event [STUDENT -> A, COURSE -> X]</li>
      *     <li> Event [STUDENT -> A, COURSE -> X, FACULTY -> 1]</li>
-     *     <li> Event [STUDENT -> A, COURSE -> Z]</li>
-     *     <li> Event [STUDENT -> B, COURSE -> Z]</li>
-     *     <li> Event [COURSE -> Z, FACULTY -> 1]</li>
      * </ul>
      * But the following events do not:
      * <ul>
-     *     <li> Event [STUDENT -> B, COURSE -> X]</li>
      *     <li> Event [STUDENT -> A]</li>
+     *     <li> Event [STUDENT -> A, COURSE -> Z]</li>
+     *     <li> Event [STUDENT -> B, COURSE -> Z]</li>
+     *     <li> Event [STUDENT -> B, COURSE -> X]</li>
      *     <li> Event [STUDENT -> Z]</li>
      * </ul>
+     * <p>
+     * Note that constructing a {@link FilteredEventCriteria} makes most sense when
+     * {@link EventStoreTransaction#source(SourcingCondition) sourcing} entities from events.
+     * For example, when sourcing events for an Aggregate, the criteria could be constructed as follows:
      *
-     * @param tags The {@link Set} of {@link Tag Tags} to compare with {@code this EventCriteria} its {@link #tags()}.
-     * @return {@code true} if they are deemed to be equal, {@code false} otherwise.
+     * <pre>
+     *     {@code
+     *     EventCriteria criteria = EventCriteria.match()
+     *     .eventTypes("StudentRegistered", "CourseRegistered")
+     *     .withTags(Tag.of("studentId", "A"))
+     *    }
+     *    </pre>
+     *
+     * @return A builder to construct a {@link FilteredEventCriteria} instance.
      */
-    default boolean matchingTags(@Nonnull Set<Tag> tags) {
-        return tags.containsAll(this.tags());
+    static EventCriteriaBuilderEventTypeStage match() {
+        return EventCriteriaBuilder.match();
     }
 
     /**
-     * Indicates whether the given {@code type} and {@code tags} matches the types and tags defined in this instance. If
-     * no types are defined, any given {@code type} will be considered a match.
+     * Create an {@code EventCriteria} that matches events that match either this {@code EventCriteria} or the given
+     * {@code EventCriteria}.
+     *
+     * @param criteria The {@code EventCriteria} to match in addition to this {@code EventCriteria}.
+     * @return An {@code EventCriteria} that matches events that match either this {@code EventCriteria} or the given
+     * {@code EventCriteria}.
+     */
+    EventCriteria or(EventCriteria criteria);
+
+
+    /**
+     * Start a builder to construct an additional {@link FilteredEventCriteria} instance that when constructed matches
+     * both the events as defined by the builder and the events as defined by the current criteria. Once construction is
+     * complete, returns an {@link OrEventCriteria} that matches events that match either this {@code EventCriteria} or
+     * the built one. See {@link EventCriteriaBuilder} for more details.
```

Comment:
I think a slight rewording makes is nicer to read:
```suggestion
     * complete, returns an {@link OrEventCriteria} that matches events from either {@code this EventCriteria} or
     * the built one. See {@link EventCriteriaBuilder} for more details.
```